contracts/cw721-base/src/lib.rs

```diff
@@ -25,133 +16,93 @@ pub use cw_ownable::{Action, Ownership, OwnershipError};
 
 use cosmwasm_std::Empty;
 
-// This is a simple type to let us handle empty extensions
-pub type Extension = Option<Empty>;
-
 // Version info for migration
 pub const CONTRACT_NAME: &str = "crates.io:cw721-base";
 pub const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
 
-// currently we only support migrating from 0.16.0. this is ok for now because
-// we have not released any 0.16.x where x != 0
-//
-// TODO: parse semvar so that any version 0.16.x can be migrated from
-pub const EXPECTED_FROM_VERSION: &str = "0.16.0";
-
 pub mod entry {
+
     use super::*;
 
     #[cfg(not(feature = "library"))]
     use cosmwasm_std::entry_point;
     use cosmwasm_std::{Binary, Deps, DepsMut, Env, MessageInfo, Response, StdResult};
+    use cw721::{
+        error::Cw721ContractError,
+        execute::Cw721Execute,
+        msg::{Cw721ExecuteMsg, Cw721InstantiateMsg, Cw721MigrateMsg, Cw721QueryMsg},
+        query::Cw721Query,
+        state::{DefaultOptionCollectionInfoExtension, DefaultOptionMetadataExtension},
+    };
 
     // This makes a conscious choice on the various generics used by the contract
     #[cfg_attr(not(feature = "library"), entry_point)]
     pub fn instantiate(
         deps: DepsMut,
         env: Env,
         info: MessageInfo,
-        msg: InstantiateMsg,
-    ) -> Result<Response, ContractError> {
-        cw2::set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
-
-        let tract = Cw721Contract::<Extension, Empty, Empty, Empty>::default();
```

Comment:
renamed: tract -> contract
never understood why it was called `tract`?!

Reply:
Hahaha, yes, contract is much clearer.